### ISSUE TITLE
Crash on the chinese Win10 with multimarkdown-6

### ISSUE THREAD

**Issue by alphatan** [jrblevin/markdown-mode#319]
When 'export', 'preview' a md file with markdown-mode with multimarkdown-6, it will consume all the memory then crash the OS (Windows)!

But running the multimarkdown command directly with the same file works fine.

## Expected Behavior

At least it should not crash the OS.

## Actual Behavior

The **multimarkdown** process consumes all the memory (16G) and crashes the OS (Win7, Win10).

## Steps to Reproduce

1. Running on Chinese Win7 or Win10
1. Installed multimarkdown-6 (6.3.0)
1. Load markdown-mode with following configurations
    ``` lisp
    (require 'markdown-mode)
    (setq markdown-command "multimarkdown") ;
    (autoload 'gfm-mode "markdown-mode"
       "Major mode for editing GitHub Flavored Markdown files" t)
    (setq markdown-fontify-code-blocks-natively nil)
    (setq markdown-xhtml-header-content "about 300K JS/CSS content")
    ```
1. Open any md doc (UTF-8 encoded)
1. Do '**export**' or '**preview**'
1. Then Cpu starts flying~~ (win7 will halt without any chance, while win10 will allow some while to kill the process)

## Backtrace

No backtrace could not be found  since the OS crashed.

## Software Versions

- Markdown Mode: From Git on 2017-10-17, 2.4-dev, and 2.3 (suffered a long time)
- Emacs: 25.3.1, 25.2, 225.1-2 (x86_64 from GNU)
- OS: Win7, Win10
- Multimarkdown: 6.3.0, 6.2.3, 6.2.2, 6.2.1, 5.4.0

## Temporary Solution

1. .emacs, setting encoding for multimarkdown process
    ``` lisp
    (when (eq system-type 'windows-nt)
      (set-default 'process-coding-system-alist
        '(
          ("[pP][lL][iI][nN][kK]" gbk )
          ("[mM][uU][lL][tT][iI][mM][aA][rR][kK][dD][oO][wW][nN]" utf-8 )
          ("[cC][mM][dD][pP][rR][oO][xX][yY]" gbk . gbk)
          )))
    ```
1. markdown-mode.el, change to call the multimarkdown.exe directly.
Find the line `(call-process-region`, then change to 
    ``` lisp
    (if (eq system-type 'windows-nt)
      (call-process-region begin-region end-region
                         markdown-command nil buf nil)
      (call-process-region begin-region end-region
                         shell-file-name nil buf nil
                         shell-command-switch markdown-command))
    ```

**Comment by jrblevin** [comment]
It seems avoiding the shell is the solution. I'm not sure there is a reason we need to use the shell anyway, so perhaps we can drop the `if` conditional and just call `markdown-command` directly in all cases. Anyone see a problem with doing that?

**Comment by syohex** [comment]
> It seems avoiding the shell is the solution. I'm not sure there is a reason we need to use the shell anyway, so perhaps we can drop the `if` conditional and just call `markdown-command` directly in all cases. Anyone see a problem with doing that?

I suppose we can use `markdown-command` directly. It was introduced by me. I cannot remember in detail, I suppose original code used `shell-command-on-region` so I replaced it with `sh -c`.

BTW I cannot reproduce this issue with Japanese UTF-8 text on Windows 10. (I don't have Windows Chinese version)

**Comment by syohex** [comment]
I confirmed using markdown-command directly works well both on Windows and Linux

**Comment by syohex** [comment]
I have merged #471. Please check latest version if you still have issue, please reopen this issue.
